src/shared/sol-conffile.c

```diff
@@ -58,87 +60,207 @@ sol_conffile_entry_sort_cb(const void *data1, const void *data2)
+            return -ENOMEM;
+            break;
+        }
+        sol_ptr_vector_append(&vec_options, tmp);
+    }
+    if (reason != SOL_JSON_LOOP_REASON_OK) {
+        int i;
+        void *ptr;
+        SOL_DBG("Error: Invalid Json.");
+        SOL_PTR_VECTOR_FOREACH_IDX(&vec_options, ptr, i) {
+            free(ptr);
+        }
+        sol_ptr_vector_clear(&vec_options);
+        sol_free_conffile_entry(entry);
+        return -ENOKEY;
+    }
 
-    groups = g_key_file_get_groups(keyfile, NULL);
-    for (groups_ptr = groups; *groups_ptr; groups_ptr++) {
-        entry = calloc(1, sizeof(*entry));
-        SOL_NULL_CHECK(entry, pv);
+    sol_ptr_vector_append(&vec_options, NULL);
+    entry->options = vec_options.base.data;
+    return 0;
+}
 
-        entry->id = strdup(*groups_ptr);
+static char *
+sol_dup_json_str(struct sol_json_token token)
+{
+    char *ret = sol_json_token_get_size(&token) >= 2 ?
+        strndup(token.start + 1, sol_json_token_get_size(&token) - 2) : NULL;
 
-        entry->type = g_key_file_get_string(keyfile,
-            *groups_ptr, "Type", NULL);
+    if (!ret)
+        SOL_WRN("Error alocating memory for string");
+    return ret;
+}
 
-        if (!entry->type) {
-            SOL_DBG("could not find mandatory 'Type' key on group [%s],"
-                " skipping entry", *groups_ptr);
-            free((void *)entry->id);
-            free(entry);
-            continue;
+static int
+sol_conffile_json_to_vector(struct sol_json_scanner scanner, struct sol_ptr_vector *pv)
+{
+    struct sol_conffile_entry *entry = NULL;
+    const char *node_group = "nodetypes";
+    const char *node_name = "name";
+    const char *node_type = "type";
+    const char *node_options = "options";
+    struct sol_json_token token, key, value;
+    struct sol_json_scanner obj_scanner;
+    enum sol_json_loop_reason reason;
+    bool found_nodes = false;
+    int i;
+
+    sol_ptr_vector_init(pv);
+
+    SOL_JSON_SCANNER_OBJECT_LOOP (&scanner, &token, &key, &value, reason) {
+        if (sol_json_token_str_eq(&key, node_group, strlen(node_group))) {
+            found_nodes = true;
+            break;
         }
+    }
+    if (reason != SOL_JSON_LOOP_REASON_OK) {
+        SOL_DBG("Error: Invalid Json.");
+        goto err;
+    }
 
-        entry->options = (const char **)g_key_file_get_string_list(keyfile,
-            *groups_ptr, "Options", NULL, NULL);
+    if (!found_nodes)
+        return -ENOKEY;
 
-        sol_ptr_vector_insert_sorted(&pv, entry,
-            sol_conffile_entry_sort_cb);
+    sol_json_scanner_init_from_token(&obj_scanner, &value);
+    SOL_JSON_SCANNER_ARRAY_LOOP (&obj_scanner, &token, SOL_JSON_TYPE_OBJECT_START, reason) {
+        entry = calloc(1, sizeof(*entry));
+        SOL_NULL_CHECK(entry, -ENOMEM);
+        SOL_JSON_SCANNER_OBJECT_LOOP_NEST (&obj_scanner, &token, &key, &value, reason) {
+            if (reason != SOL_JSON_LOOP_REASON_OK) {
+                SOL_DBG("Error: Invalid Json.");
```

Comment:
s/Json/JSON/. JSON is an acronym (JavaScript Object Notation), and all acronyms should be written in capital letters.